libsolidity/interface/CompilerStack.cpp

```diff
@@ -649,55 +674,85 @@ bool CompilerStack::compile(State _stopAfter)
 	// Only compile contracts individually which have been requested.
 	map<ContractDefinition const*, shared_ptr<Compiler const>> otherCompilers;
 
-	for (Source const* source: m_sourceOrder)
-		for (ASTPointer<ASTNode> const& node: source->ast->nodes())
-			if (auto contract = dynamic_cast<ContractDefinition const*>(node.get()))
-				if (isRequestedContract(*contract))
-				{
-					try
+	if (!m_evmAssemblyJson.empty())
+	{
```

Comment:
Can this if-body be somehow be combined with (some part of) `CompilerStack::assemble(...)`?

Reply:
I guess that would be possible.. at least for now `CompilerStack::compile(...)` is right now not calling `CompilerStack::assemble(...)` - but I guess it would be better if the whole logic could be somehow moved to `CompilerStack::assemble(...)`. Will look into this later.

Reply:
Looking at this again. I think it is ok how this is done right now. I don't see anything that could be moved into the `CompilerStack::assemble(...)` method.